persistent-postgresql/Database/Persist/Postgresql.hs

```diff
@@ -1311,6 +1313,27 @@ repsertManySql ent n = putManySql' conflictColumns fields ent n
     fields = keyAndEntityFields ent
     conflictColumns = escape . fieldDB <$> entityKeyFields ent
 
+insertUniqueSql :: EntityDef -> [PersistValue] -> InsertSqlResult
+insertUniqueSql ent vals =
+    let sql = T.concat
+            [ "INSERT INTO "
+            , escape $ entityDB ent
+            , if null (entityFields ent)
+                then " DEFAULT VALUES"
+                else T.concat
+                    [ "("
+                    , T.intercalate "," $ map (escape . fieldDB) $ entityFields ent
+                    , ") VALUES("
+                    , T.intercalate "," $ map (const "?") $ entityFields ent
+                    , ") ON CONFLICT ("
+                    , T.intercalate "," $ concat $ map (\x -> map escape (map snd $ uniqueFields x)) (entityUniques ent)
```

Comment:
This is also not taking into account the `entityId`, so in the test suite example, we're not catching a conflict on `foo`, the `Primary` key.